tests/rptest/tests/shadow_indexing_compacted_topic_test.py

```diff
@@ -106,6 +123,19 @@ def compacted_segments_uploaded():
                    backoff_sec=2,
                    err_msg=lambda: f"Compacted segments not uploaded")
 
+        # check that the metric is increasing. the check needs to account for leadership changes (unlikely in this test),
+        # because the current implementation does not persist the value in the manifest or the log
+        compacted_replaced_bytes_increasing = m.evaluate([
+            (metric, lambda old, new: old < new)
+        ])
+        new_topic_describe = list(self._rpk_client.describe_topic(
```

Comment:
In my experience this could be flaky because rpk can return an empty list in some cases. This could be fixed by using `wait_until`.